### PR TITLE
synology web UI

### PR DESCRIPTION
**Pull request by crawshaw**
With this the web UI login button works on DSM 6.

### CODE REVIEW

File: cmd/tailscale/cli/web.go [hunk 1]
```diff
@@ -259,6 +265,9 @@ func tailscaleUp(ctx context.Context) (authURL string, retErr error) {
 	c, bc, ctx, cancel := connect(ctx)
 	defer cancel()
 
+	var loginOnce sync.Once
+        startLoginInteractive := func() { loginOnce.Do(func() { bc.StartLoginInteractive() }) }
```
Comment (bradfitz) [inline cmd/tailscale/cli/web.go:269]: gofmt

File: paths/paths.go [hunk 2]
```diff
@@ -36,6 +36,11 @@ func DefaultTailscaledSocket() string {
 	if runtime.GOOS == "darwin" {
 		return "/var/run/tailscaled.socket"
 	}
+	// TODO(crawshaw): does this path change with DSM7?
+	const synologySock = "/volume1/@appstore/Tailscale/var/tailscaled.sock" // SYNOPKG_PKGDEST in scripts/installer
```
Comment (bradfitz) [inline paths/paths.go:40]: Does this work even for side-loaded SPKs, or only via the `@appstore`?

Comment (crawshaw) [reply paths/paths.go:40]: It does on DSM6. I suspect there will be changes coming soon for DSM7.

File: paths/paths.go [hunk 3]
```diff
@@ -26,6 +26,11 @@ func DefaultTailscaledSocket() string {
...
+	if fi, err := os.Stat(filepath.Dir(synologySock)); err == nil && fi.IsDir() {
```
Comment (bradfitz) [inline paths/paths.go:31]: can you at least do this conditionally on runtime.GOOS == "linux" so we don't do all these stats on other operating systems?

**Comment by crawshaw** [comment]
OK this code is now in line with how 'tailscale up' now works.

**Review by bradfitz — APPROVED** [review]

**Pull request merged**